File: internal/content/testdata/bench/java/nc-linked-list-cycle.java
class ListNode {
    int val;
    ListNode next;
    ListNode(int val) { this.val = val; }
}

class Solution {
    public boolean has_cycle(long[] values, long pos) {
        if (values == null || values.length == 0) return false;
        ListNode[] nodes = new ListNode[values.length];
        for (int i = 0; i < values.length; i++) {
            nodes[i] = new ListNode((int) values[i]);
        }
        for (int i = 0; i < nodes.length - 1; i++) {
            nodes[i].next = nodes[i + 1];
        }
        if (pos >= 0 && nodes.length > 0) {
            nodes[nodes.length - 1].next = nodes[(int) pos];
        }
        ListNode slow = nodes[0], fast = nodes[0];
        while (fast != null && fast.next != null) {
            slow = slow.next;
            fast = fast.next.next;
            if (slow == fast) return true;
        }
        return false;
    }
}

File: internal/content/testdata/bench/csharp/nc-linked-list-cycle.cs
public class Solution {
    private class CycleNode {
        public long val;
        public CycleNode next;
        public CycleNode(long v) { val = v; }
    }

    public bool has_cycle(long[] values, long pos) {
        if (values == null || values.Length == 0) return false;
        var nodes = new CycleNode[values.Length];
        for (int i = 0; i < values.Length; i++) nodes[i] = new CycleNode(values[i]);
        for (int i = 0; i < values.Length - 1; i++) nodes[i].next = nodes[i + 1];
        if (pos >= 0) nodes[values.Length - 1].next = nodes[(int)pos];

        var slow = nodes[0]; var fast = nodes[0];
        while (fast != null && fast.next != null) {
            slow = slow.next;
            fast = fast.next.next;
            if (slow == fast) return true;
        }
        return false;
    }
}
